arrow-integrations-retrofit-adapter/src/test/kotlin/arrow/integrations/retrofit/adapter/TaglessCallTest.kt

```diff
@@ -0,0 +1,92 @@
+package arrow.integrations.retrofit.adapter
+
+import arrow.core.right
+import arrow.core.test.UnitSpec
+import arrow.fx.reactor.MonoK
+import arrow.fx.reactor.extensions.monok.async.async
+import arrow.fx.reactor.unsafeRunSync
+import arrow.fx.rx2.ObservableK
+import arrow.fx.rx2.extensions.observablek.async.async
+import arrow.integrations.retrofit.adapter.io.TaglessAdapterFactory
+import arrow.integrations.retrofit.adapter.mock.ResponseMock
+import arrow.integrations.retrofit.adapter.retrofit.IOApiClientTest
+import arrow.integrations.retrofit.adapter.retrofit.MonoKApiClientTest
+import arrow.integrations.retrofit.adapter.retrofit.ObservableKApiClientTest
+import arrow.integrations.retrofit.adapter.retrofit.retrofit
+import io.kotlintest.fail
+import okhttp3.HttpUrl
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Assert.assertEquals
+
+class TaglessCallTest : UnitSpec() {
+
+  private fun server(): MockWebServer = MockWebServer().apply {
+    enqueue(MockResponse().setBody("{\"response\":  \"hello, world!\"}").setResponseCode(200))
+    start()
+  }
+
+  private fun baseUrl(): HttpUrl = server().url("/")
+
+  init {
+    "should be able to parse answer with IO" {
+      createIOApiClientTest(baseUrl())
+        .testIO()
+        .unsafeRunSyncEither()
+        .fold({ throwable ->
+          fail("The requested terminated with an exception. Message: ${throwable.message}")
+        }, { responseMock ->
+          assertEquals(ResponseMock("hello, world!"), responseMock)
+        })
+    }
+
+    "should be able to parse answer with ObservableK" {
+        createObservableKApiClientTest(baseUrl())
+          .testObservableK()
+          .observable
+          .test()
+          .await()
+          .assertValue { it == ResponseMock("hello, world!") }
+      }
+
+  "should be able to parse answer with MonoK" {
+    val result = createMonoKApiClientTest(baseUrl())
+      .testMonoK()
+      .unsafeRunSync()
+    assertEquals(result, ResponseMock("hello, world!"))
+  }
+
+    "should be able to run IO POST with UNIT as response" {
+      val result = createIOApiClientTest(baseUrl())
+        .testUnitResponsePost()
+        .unsafeRunSyncEither()
+      assertEquals(result, Unit.right())
+    }
+
+    "should be able to run ObservableK POST with UNIT as response" {
+      createObservableKApiClientTest(baseUrl())
+        .testUnitResponsePost()
+        .observable
+        .test()
+        .await()
+        .assertValue { it == Unit }
+    }
+
+    "should be able to run MonoK POST with UNIT as response" {
+      val result = createMonoKApiClientTest(baseUrl())
+        .testUnitResponsePost()
+        .unsafeRunSync()
+
+      assertEquals(result, Unit)
+    }
+  }
```

Comment:
Ideally the tests should also check the error cases (2 or 3 depending on the null discussion) so all scenarios are covered